packages/components/src/components/DetailsSearch.vue

```diff
@@ -197,6 +204,24 @@ export default {
       } else {
         this.$store.commit(SELECT_CODE_OBJECT, object);
       }
+
+      if (type === 'analysis-finding') this.$emit('analysisFindingSelection', object);
+    },
+
+    analysisFindingSelection(findingObject) {
+      const finding = findingObject.resolvedFinding && findingObject.resolvedFinding.finding;
+      if (!finding) return;
+
+      if (finding.checkId === 'n-plus-one-query') {
```

Comment:
Is this just for prototype purposes? Hard coding diagram behavior for specific findings is probably not necessary if we just base the behavior on the Finding data / type.